src/AutoRest.CSharp/Common/Generation/Writers/ModelWriter.cs

```diff
@@ -124,9 +124,7 @@ private void WriteProperty(CodeWriter writer, ObjectTypeProperty property, Objec
             writer.WriteXmlDocumentationSummary(CreatePropertyDescription(property));
             if (Configuration.EnableBicepSerialization && objectType.Declaration.Accessibility == "public" && property.Declaration.Accessibility == "public")
             {
-                string? wrapper = property.SchemaProperty?.FlattenedNames?.FirstOrDefault();
-                wrapper = wrapper is null ? string.Empty : $"{wrapper}.";
-                writer.Line($"[WirePath(\"{wrapper}{property.SerializedName}\")]");
+                writer.Line($"[WirePath(\"{property.GetWirePath()}\")]");
```

Comment:
Left this change in but the wirepath isn't needed for the changes in this PR. @m-nash let me know if I should remove it.